### PR TITLE
refactor(TabSwitcher): move TabSwitcher from class to function

### PR DESCRIPTION
**Pull request by spartDev**
- [ ] Feature
- [ ] Fix
- [x] Enhancement

## Description

Move TabSwitcher from class to function
React hooks are awesome!

## Todo - Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.

### CODE REVIEW

File: src/TabSwitcher/README.md [hunk 1]
```diff
@@ -10,17 +10,19 @@ import { TabSwitcher } from '@tillersystems/stardust';
...
-| propName            | propType  | defaultValue | isRequired |
-| ------------------- | :-------: | :----------: | :--------: |
-| `activeIndex`       | `number`  |     `0`      |     -      |
-| `children`          |  `node`   |    `null`    |     -      |
-| `isCompacted`       | `boolean` |   `false`    |     -      |
-| `onActiveTabChange` |  `func`   |  `() => {}`  |     -      |
+- `children` - Tabs and Panes to be displayed.
+- `defaultIndex` - Starts the tab at a specific index.
+- `index` - Like form inputs, a tab's state can be controlled by the owner.
+- `isCompacted` - If it true should reduce its size by reducing padding and font-size.
```
Comment (sun-tea) [inline src/TabSwitcher/README.md]: `If at true,` or `If it is true,`

File: src/TabSwitcher/__stories__/TabSwitcher.stories.js [hunk 2]
```diff
@@ -38,8 +40,8 @@ storiesOf('TabSwitcher', module)
       },
     ];
 
-    const activeIndexValue = number(
-      'activeIndex',
+    const degaultIndexValue = number(
```
Comment (sun-tea) [inline src/TabSwitcher/__stories__/TabSwitcher.stories.js]: `defaultIndexValue`

File: src/TabSwitcher/index.js [hunk 3]
```diff
@@ -12,85 +12,66 @@ const { bool, func, node, number } = PropTypes;
...
-    });
+TabSwitcher.propTypes = {
+  children: node.isRequired,
+  defaultIndex: number,
+  isCompacted: bool,
+  onChange: func,
+  index: (props, name, compName, ...rest) => {
+    if (props.index > -1 && props.onChange == null) {
+      return new Error(
+        'You provided a `index` prop to `TabSwitcher` without an `onChange` handler. This will render a read-only tabs element. If the tabs should be mutable use `defaultIndex`. Otherwise, set `onChange`.',
```
Comment (sun-tea) [inline src/TabSwitcher/index.js]: an index

**Comment by sun-tea** [comment]
Please squash the last commit

**Comment by spartDev** [comment]
done!

**Pull request merged**